src/connector/LeverageMacroBase.sol

```diff
@@ -0,0 +1,393 @@
+    ///     And if you really need eq, it's third
+    function _doCheckValueType(CheckValueAndType memory check, uint256 valueToCheck) internal {
+        if (check.operator == Operator.skip) {
+            // Early return
+            return;
+        } else if (check.operator == Operator.gte) {
+            require(check.value >= valueToCheck, "!LeverageMacroReference: gte post check");
+        } else if (check.operator == Operator.lte) {
+            require(check.value <= valueToCheck, "!LeverageMacroReference: let post check");
+        } else if (check.operator == Operator.equal) {
+            require(check.value == valueToCheck, "!LeverageMacroReference: equal post check");
+        } else {
+            revert("Operator not found");
+        }
+    }
+
+    /// @notice Convenience function to parse bytes into LeverageMacroOperation data
+    function decodeFLData(bytes calldata data) public view returns (address, LeverageMacroOperation memory) {
+        (address caller, LeverageMacroOperation memory leverageMacroData) = abi.decode(data, (address, LeverageMacroOperation));
+        return (caller, leverageMacroData);
+    }
+
+    /// @notice Proper Flashloan Callback handler
+    function onFlashLoan(
+        address initiator,
+        address token,
+        uint256 amount,
+        uint256 fee,
+        bytes calldata data
+    ) external returns (bytes32) {
+        // Verify we started the FL
+        require(initiator == address(this), "LeverageMacroReference: wrong initiator for flashloan");
+
+        // Ensure the caller is the intended contract
+        if (token == address(ebtcToken)) {
+            require(
+                msg.sender == address(borrowerOperations),
+                "LeverageMacroReference: wrong lender for eBTC flashloan"
+            );
+        } else {
+            // Enforce that this is either eBTC or stETH
+            require(
+                msg.sender == address(activePool),
+                "LeverageMacroReference: wrong lender for stETH flashloan"
+            );
+        }
+
+        // Else a malicious contract, that changes the data would be able to inject a forwarded caller
+
+        // Get the data
+        // We will get the first byte of data for enum an type
+        // The rest of the data we can decode based on the operation type from calldata
+        // Then we can do multiple hooks and stuff
+        (address caller, LeverageMacroOperation memory operation) = decodeFLData(data);
+
+        IERC20(token).transfer(caller, amount);
+
+        string[] memory targetNames = new string[](1);
+        bytes[] memory datas = new bytes[](1);
+
+        targetNames[0] = "Badger-Zap-Router-v1.0";
+        datas[0] = abi.encodeWithSelector(
+            IBadgerConnector.handleOperation.selector,
+            token,
+            amount,
+            fee,
+            operation
+        );
+
+        IDSAAccount(caller).cast(targetNames, datas, 0x03d70891b8994feB6ccA7022B25c32be92ee3725);
```

Comment:
TODO: What is this?

https://etherscan.io/address/0x03d70891b8994feB6ccA7022B25c32be92ee3725#tokentxns